consensus/src/validation/step.rs

```diff
@@ -73,39 +92,28 @@ impl<T: Operations + 'static> ValidationStep<T> {
         };
 
         // Call Verify State Transition to make sure transactions set is valid
-        if let Err(err) = Self::call_vst(candidate, executor).await {
-            error!(event = "failed_vst_call", ?err);
-            return;
-        }
+        let vote = match Self::call_vst(candidate, executor).await {
+            Ok(_) => Vote::Valid(header.hash),
+            Err(err) => {
+                error!(event = "failed_vst_call", ?err);
+                Vote::Invalid(header.hash)
```

Comment:
Returning `Vote::Invalid(header.hash)`  must be  also done on a failure of verify_block_header (a few lines above), correct?

Reply:
Yes, you should be definitely right. 

Ps: Does it worth to add a byte to `Vote::Invalid` to specify the reason of the failure?
It would be useful to implement different kind of slash, but it would impact the vote signature aggregation

Reply:
An extra flag is out of the scope of this PR, not needed for now IMO.

Reply:
As you say, it would impact aggregation...I'm not sure it's worth it. If the block is deemed invalid, we can always check it ourselves and decide accordingly.